### ISSUE TITLE
MonkeyPatchWarning resulting in OSError: [Errno 8]

### ISSUE THREAD

**Issue by reinoutvg** [miguelgrinberg/Flask-SocketIO#928]
I've used Flask-SocketIO for quite a while now. Today I reinstalled my ubuntu and thus all my python modules. When I run my test file I receive the following message:

>  python '/home/ubuntu/Desktop/Workingfolder/TWT-app/runApp.py' 
/home/ubuntu/.local/lib/python3.6/site-packages/flask_socketio/__init__.py:557: MonkeyPatchWarning: Monkey-patching ssl after ssl has already been imported may lead to errors, including RecursionError on Python 3.6. It may also silently lead to incorrect behaviour on Python 3.7. Please monkey-patch earlier. See https://github.com/gevent/gevent/issues/1016. Modules that had direct imports (NOT patched): ['urllib3.util.ssl_ (/home/ar-ubuntu/.local/lib/python3.6/site-packages/urllib3/util/ssl_.py)', 'urllib3.util (/home/ar-ubuntu/.local/lib/python3.6/site-packages/urllib3/util/__init__.py)']. 
  monkey.patch_all()

resulting in the following error:

> OSError: [Errno 8] Exec format error: '/home/ubuntu/Desktop/Workingfolder/app/runApp.py'

Any help for handling this error? Thanks.

**Comment by miguelgrinberg** [comment]
This means that you need to monkey patch before you do everything else. Move the monkey patching to the top of the main script, even above all other imports.

**Comment by reinoutvg** [comment]
I was stuck on this issue all day. Until today I never heard of monkey patching and I can't really find what I have to move where. However, I thinking that the gevent-websocket is not communicating really well with flask_socketio on my machine for some reason.

This is what I've done so far:
- pip uninstall gevent-websocket
- pip uninstall gevent
- pip uninstall python-socketio
- pip install python-socketio
- added #!/usr/bin/env python to the top of my python file. 

One step forward although I'm having this error now:

> WebSocket transport not available. Install eventlet or gevent and gevent-websocket for improved performance.
 * Serving Flask app "runApp" (lazy loading)
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: on
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
 * Restarting with stat
Traceback (most recent call last):
  File "/home/ubuntu/Desktop/Workingfolder/app/runApp.py", line 5, in <module>
    from flask_socketio import SocketIO, emit, send
ImportError: No module named flask_socketio

I checked on my disk and the module is there for sure. 
I checked this post: https://github.com/miguelgrinberg/Flask-SocketIO/issues/164. Although there's no answer for me here. 

I'm stuck now, any help is appreciated.

**Comment by miguelgrinberg** [comment]
You need to install `flask-socketio`.

**Comment by reinoutvg** [comment]
Already did of course.
Like I said I checked on my disk and the module is there for sure and also in the right folder.
Also when I check pip freeze i says Flask-SocketIO==3.3.2.

**Comment by miguelgrinberg** [comment]
how do you start your application?

**Comment by reinoutvg** [comment]
Turned out to be that the runApp.py file that I copied from my server to my new Ubuntu drive (because of the reinstall of Ubuntu) became dirty, un-trustful or whatever. 

All steps together this was it for me:
pip uninstall gevent-websocket
pip uninstall gevent
pip uninstall python-socketio
pip install python-socketio
started a py file from scratch
inserted the example of the flask SocketIO on the main page and build it back to my previous file

And it worked fine. Thanks for the help.
